packages/ui/src/components/popover/popover.tsx

```diff
@@ -0,0 +1,114 @@
+import { h, Component } from "preact";
+import * as styles from "./popover.scss";
+
+const VERTICAL_POSITIONS = ["top", "bottom"];
+const HORIZONTAL_POSITIONS = ["left", "right"];
+const TRIGGER_MOD = ["click", "hover"];
+
+const defaultProps = {
+    verticalPosition: VERTICAL_POSITIONS[0],
+    horizontalPosition: HORIZONTAL_POSITIONS[0],
+    triggerMod: TRIGGER_MOD[0],
+    className: "popover",
+    open: false,
+    closeOnEsc: true
+};
+
+interface PopoverProps {
+    onClose?: () => void;
+    onOpen?: () => void;
```

Comment:
those two properties together with `open` property indicate that the parent of this component controlling on this component open state. But once you provided `triggerMode` property you actually singling others that this component is on charge on the open state. It should be either or:
1. either the parent control -> meaning you are not the one who render the `anchorEl` and you have `open` prop only instead of the `onOpen, onClose, triggerMode` properties
2. or you are in control -> meaning you remove `onOpen, onClose, open` , add internal state to keep track on the `open` mode.

the second option is much more popular so let's try to achieve that one.

also, asking `anchorEl` indicate that someone else render that component. in your implementation you are handling the rendering. As such you cannot name it `anchorEl` and `children` are not necessary the content.

this is taken from [ant design](https://ant.design/components/popover/).

let's do the same. make the children as our visible component and `content` property as the content to show as part of the popover
```
import { Popover, Button } from 'antd';

const content = (
  <div>
    <p>Content</p>
    <p>Content</p>
  </div>
);

ReactDOM.render(
  <Popover content={content} title="Title">
    <Button type="primary">Hover me</Button>
  </Popover>,
  mountNode,
);
```